src/audio/kpb.c

```diff
@@ -713,17 +713,13 @@ static void kpb_init_draining(struct comp_data *kpb, struct kpb_client *cli)
 	if (cli->id > KPB_MAX_NO_OF_CLIENTS) {
 		trace_kpb_error("kpb_init_draining() error: "
 				"wrong client id");
-		return;
```

Comment:
Why do you remove instant returns here? If you don't return the init draining procedure will continue and probably will schedule draining task even though it should not happen due to wrong request.

Reply:
@mrajwa Isn't the draining task in the last else statement?

Reply:
@tlauda, yes it is fine then.